CONTRIBUTING.rst

```diff
@@ -256,8 +263,54 @@ When developing, you can make Python changes available in R by running (from mlf
 Please also follow the recommendations from the
 `Advanced R - Style Guide <http://adv-r.had.co.nz/Style.html>`_ regarding naming and styling.
 
+Java
+~~~~
+
+If contributing to MLflow's Java APIs or modifying Java documentation,
+install `Java <https://www.java.com/>`_ and `Apache Maven <https://maven.apache.org/download.cgi>`_.
+
+Certain MLflow modules are implemented in Java, under the ``mlflow/java/`` directory.
+These are the Java Tracking API client (``mlflow/java/client``) and the Model Scoring Server
+for Java-based models like MLeap (``mlflow/java/scoring``).
+
+Other Java functionality (like artifact storage) depends on the Python package, so first install
+the Python package in a conda environment as described above.
+`Install <https://www.oracle.com/technetwork/java/javase/downloads/index.html>`_
+the Java 8 JDK (or above), and `download <https://maven.apache.org/download.cgi>`_
+and `install <https://maven.apache.org/install.html>`_ Maven. You can then build and run tests via:
+
+.. code-block:: bash
+
+  cd mlflow/java
+  mvn compile test
+
+If opening a PR that makes API changes, please regenerate API documentation as described in
+`Writing Docs`_ and commit the updated docs to your PR branch.
+
 Python
 ~~~~~~
+If you are contributing in Python, make sure that you have satisfied all the `General prerequisites and dependencies`_,
+including installing ``pytest``, as you will need it for the sections described below.
+
+Writing Python Tests
+++++++++++++++++++++
+If your PR includes code that isn't currently covered by our tests (e.g. adding a new flavor, adding
+autolog support to a flavor, etc.), you should write tests that cover your new code. MLflow currently
```

Comment:
Can we delete the section "MLflow currently uses ``pytest==3.2.1`` for testing?" Seems unlikely to be immediately useful to readers of this guide & hard to keep up-to-date

Reply:
Okay, done!